### PR TITLE
feat: add allocate cash component

### PR DESCRIPTION
**Pull request by BLasan**
## Description
Allocate Cash component created.

## Related issues and discussion
#758 

## Screenshots, if any
Screenshot from 2020-06-16 15-46-44

## Checklist
Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] If you have multiple commits please combine them into one commit by squashing them.

- [x] Read and understood the contribution guidelines at `web-app/.github/CONTRIBUTING.md`.

### CODE REVIEW

**Comment by BLasan** [comment]
@karantakalkar  Please review

File: src/app/organization/tellers/allocate-cash/allocate-cash.component.ts [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    this.allocateCashForm.patchValue({
+      txnDate: this.datePipe.transform(txnDate, dateFormat)
+    });
+    const allocateCashForm = this.allocateCashForm.value;
+    allocateCashForm.dateFormat = dateFormat;
+    allocateCashForm.locale = 'en';
+    delete allocateCashForm.tellerName;
+    delete allocateCashForm.office;
+    delete allocateCashForm.cashier;
+    delete allocateCashForm.assignmentPeriod;
```
Comment (karantakalkar) [inline src/app/organization/tellers/allocate-cash/allocate-cash.component.ts:82]: Is this necessary? I guess disabled controls are not included in form value, can you check?

File: src/app/organization/tellers/allocate-cash/allocate-cash.component.ts [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+  /**
+   * Set Cashier form.
+   */
+  setCashierForm() {
+    this.allocateCashForm = this.formBuilder.group({
+      'office': [{value: this.cashierData.officeName, disabled: true}],
+      'tellerName': [{value: this.cashierData.tellerName, disabled: true}],
+      'cashier': [{value: this.cashierData.cashierName, disabled: true}],
+      'assignmentPeriod': [{value: this.cashierData.startDate && new Date(this.cashierData.startDate), disabled: true}],
```
Comment (karantakalkar) [inline src/app/organization/tellers/allocate-cash/allocate-cash.component.ts:59]: Assignment period is from start date to end date, `start date - end date`

File: src/app/organization/tellers/allocate-cash/allocate-cash.component.html [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+            </mat-error>
+          </mat-form-field>
+
+          <mat-form-field fxFlex="48%">
+            <mat-label>Currency</mat-label>
+            <mat-select required formControlName="currencyCode">
+              <mat-option *ngFor="let currency of cashierData.currencyOptions" [value]="currency.code">
+                {{ currency.name }}
+              </mat-option>
+            </mat-select>
```
Comment (karantakalkar) [inline src/app/organization/tellers/allocate-cash/allocate-cash.component.html:47]: mat error missing

File: src/app/organization/tellers/allocate-cash/allocate-cash.component.html [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+    <form [formGroup]="allocateCashForm" (ngSubmit)="submit()">
+
+      <mat-card-content>
+
+        <div fxLayout="row wrap" fxLayoutGap="2%" fxLayout.lt-md="column">
+
+          <mat-form-field fxFlex="48%">
+            <mat-label>Office</mat-label>
+            <input matInput required autofocus formControlName="office">
```
Comment (karantakalkar) [inline src/app/organization/tellers/allocate-cash/allocate-cash.component.html:13]: what is autofocus used for?

File: src/app/organization/tellers/allocate-cash/allocate-cash.component.ts [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+export class AllocateCashComponent implements OnInit {
+
+  /** Cashier data. */
+  cashierData: any;
+  /** Cashier Form. */
+  allocateCashForm: FormGroup;
+  /** Minimum Date allowed. */
+  minDate = new Date(2000, 0, 1);
+  /** Maximum Date allowed. */
+  maxDate = new Date();
```
Comment (karantakalkar) [inline src/app/organization/tellers/allocate-cash/allocate-cash.component.ts:27]: shift above cashierdata and form variables

File: src/app/organization/tellers/allocate-cash/allocate-cash.component.ts (hunk 2 above)
Comment (BLasan) [reply src/app/organization/tellers/allocate-cash/allocate-cash.component.ts:59]: Isn't it the start date?

File: src/app/organization/tellers/allocate-cash/allocate-cash.component.html (hunk 3 above)
Comment (BLasan) [reply src/app/organization/tellers/allocate-cash/allocate-cash.component.html:50]: Done

File: src/app/organization/tellers/allocate-cash/allocate-cash.component.ts (hunk 5 above)
Comment (BLasan) [reply src/app/organization/tellers/allocate-cash/allocate-cash.component.ts:27]: Please check

File: src/app/organization/tellers/allocate-cash/allocate-cash.component.ts (hunk 2 above)
Comment (karantakalkar) [reply src/app/organization/tellers/allocate-cash/allocate-cash.component.ts:59]: no see ui in commy app

**Comment by BLasan** [comment]
@karantakalkar Done. Please review

File: src/app/organization/tellers/allocate-cash/allocate-cash.component.html (hunk 3 above)
Comment (karantakalkar) [reply src/app/organization/tellers/allocate-cash/allocate-cash.component.html:50]: You have placed it within mat select

File: src/app/organization/tellers/allocate-cash/allocate-cash.component.ts (hunk 5 above)
Comment (karantakalkar) [reply src/app/organization/tellers/allocate-cash/allocate-cash.component.ts:27]: place min and max date as first 2 variables

File: src/app/organization/tellers/allocate-cash/allocate-cash.component.html (hunk 3 above)
Comment (BLasan) [reply src/app/organization/tellers/allocate-cash/allocate-cash.component.html:47]: Resoved

File: src/app/organization/tellers/allocate-cash/allocate-cash.component.ts (hunk 5 above)
Comment (BLasan) [reply src/app/organization/tellers/allocate-cash/allocate-cash.component.ts:23]: Resolved

**Comment by BLasan** [comment]
@karantakalkar Please review

**Comment by karantakalkar** [comment]
@BLasan resolve please

**Comment by BLasan** [comment]
> @BLasan resolve please

Done

**Comment by karantakalkar** [comment]
fix lint errors

**Comment by BLasan** [comment]
> fix lint errors

Done

**Pull request merged**